packages/components-native/src/InputFile/InputFile.tsx

```diff
@@ -0,0 +1,655 @@
+const imageQualityOptions: Record<
+  Required<InputFileProps>["imageQuality"],
+  ResizeOptions | undefined
+> = {
+  original: {
+    maxWidth: undefined,
+    maxHeight: undefined,
+    quality: undefined,
+  },
+  email: {
+    maxWidth: EMAIL_IMAGE_DIMENSION,
+    maxHeight: EMAIL_IMAGE_DIMENSION,
+    quality: 0.7,
+  },
+  small: {
+    maxWidth: SMALL_IMAGE_DIMENSION,
+    maxHeight: SMALL_IMAGE_DIMENSION,
+    quality: 1,
+  },
+};
+
+// Using MIME types and partial matches to include most files but
+//  not image or video
+//  reference: https://www.iana.org/assignments/media-types/media-types.xhtml
+const restrictedToTypes = ["application/*", "audio/*", "text/*"];
+
+interface FileToUpload {
+  sourceFile: SourceFile;
+  fileUpload: FileUpload;
+}
+
+type MediaSource = FileSource.gallery | FileSource.video | FileSource.camera;
+
+interface MediaSourceSettings {
+  readonly requestPermission: () => Promise<PermissionsType>;
+  readonly permitted: (permissions: PermissionsType) => boolean;
+  readonly capture: (
+    responseHandler: (response: ImagePickerResponse) => void,
+  ) => void;
+  readonly messages: {
+    permissionDeniedTitle: typeof messages.permissionFailedCamera;
+    permissionDeniedMessage: typeof messages.permissionFailedDescription;
+  };
+}
+
+const mediaTypes: Record<AllowedTypes, MediaType> = {
+  images: "photo",
+  videos: "video",
+  mixed: "mixed",
+};
+
+function useInputFileHooks() {
+  const { onLogError, isOnline } = useAtlantisContext();
+  return { isOnline, onLogError };
+}
+
+const IS_ANDROID_33_OR_GREATER =
+  Platform.OS === "android" && Platform.Version >= 33;
+
+function getMenuOptions(formatMessage: IntlShape["formatMessage"]) {
+  const menuOptions = {
+    images: [
+      {
+        title: formatMessage(messages.takePhoto),
+        value: FileSource.camera,
+      },
+      {
+        title: formatMessage(messages.chooseFromLibrary),
+        value: FileSource.gallery,
+      },
+    ],
+    videos: [
+      ...(IS_ANDROID_33_OR_GREATER
+        ? []
+        : [
+            {
+              title: formatMessage(messages.takeVideo),
+              value: FileSource.video,
+            },
+          ]),
+      {
+        title: formatMessage(messages.chooseFromLibrary),
+        value: FileSource.gallery,
+      },
+    ],
+    mixed: [
+      {
+        title: formatMessage(messages.takePhoto),
+        value: FileSource.camera,
+      },
+      ...(IS_ANDROID_33_OR_GREATER
+        ? []
+        : [
+            {
+              title: formatMessage(messages.takeVideo),
+              value: FileSource.video,
+            },
+          ]),
+      {
+        title: formatMessage(messages.chooseFromLibrary),
+        value: FileSource.gallery,
+      },
+      {
+        title: formatMessage(messages.browseFiles),
+        value: FileSource.document,
+      },
+    ],
+  };
+
+  return menuOptions;
+}
+
+/* eslint max-statements: ["error", 15] */
```

Comment:
Needed to add this due to the console logs. We should cleanup the function definitions in the closure but I wanted to keep the scope of this PR as small as possible since it is already pretty big